src/engine/contactreader.cpp

```diff
@@ -1150,7 +1150,7 @@ static QString buildWhere(
 
         // We need to perform case-insensitive matching if MatchFixedString is specified (unless
         // CaseSensitive is also specified)
-        bool caseInsensitive = stringField && fixedString && ((filter.matchFlags() & QContactFilter::MatchCaseSensitive) == 0);
```

Comment:
does calling "lower(columnName)" cause some things not to work (e.g. numeric values to be treated as strings and then not being bound properly, when queried) perhaps?